x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -6,157 +6,143 @@
+    [
+      checkForFindings(esClient.asCurrentUser, true, logger),
       packageService.asInternalUser.getInstallation(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
       packageService.asInternalUser.fetchFindLatestPackage(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
       getCspPackagePolicies(soClient, packagePolicyService, CLOUD_SECURITY_POSTURE_PACKAGE_NAME, {
         per_page: 10000,
       }),
-    ]);
+    ]
+  );
 
   const healthyAgents = await getHealthyAgents(
     soClient,
-    installedIntegrations.items,
+    installedPackagePolicies.items,
     agentPolicyService,
     agentService
   );
 
-  const installedIntegrationsTotal = installedIntegrations.total;
-  const latestPackageVersion = latestPackageInfo.version;
+  const installedPackagePoliciesTotal = installedPackagePolicies.total;
+  const latestCspPackageVersion = latestCspPackage.version;
 
-  const status = getStatus(
-    findingsIndexExists,
-    installedIntegrationsTotal,
+  const MIN_DATE = 0;
+  const status = calculateCspStatusCode(
+    hasFindings,
+    installedPackagePoliciesTotal,
     healthyAgents,
-    getMinutesPassedSinceMoment(installationPackageInfo?.install_started_at || 0)
+    calculateDiffFromNowInMinutes(installation?.install_started_at || MIN_DATE)
   );
 
   if (status === 'not-installed')
     return {
       status,
-      latestPackageVersion,
+      latestPackageVersion: latestCspPackageVersion,
       healthyAgents,
-      installedIntegrations: installedIntegrationsTotal,
+      installedPackagePolicies: installedPackagePoliciesTotal,
     };
 
   return {
     status,
-    latestPackageVersion,
+    latestPackageVersion: latestCspPackageVersion,
     healthyAgents,
-    installedIntegrations: installedIntegrationsTotal,
-    installedPackageVersion: installationPackageInfo?.install_version,
+    installedPackagePolicies: installedPackagePoliciesTotal,
+    installedPackageVersion: installation?.install_version,
   };
 };
 
-export const defineGetCspSetupStatusRoute = (router: CspRouter): void =>
-  router.get(
-    {
-      path: STATUS_ROUTE_PATH,
-      validate: false,
-      options: {
-        tags: ['access:cloud-security-posture-read'],
-      },
-    },
-    async (context, _, response) => {
-      const cspContext = await context.csp;
-      try {
-        const cspSetupStatus = await getCspSetupStatus(
-          cspContext.logger,
-          cspContext.esClient.asCurrentUser,
-          cspContext.soClient,
-          cspContext.packageService,
-          cspContext.packagePolicyService,
-          cspContext.agentPolicyService,
-          cspContext.agentService
-        );
-
-        const body: CspSetupStatus = cspSetupStatus;
-
-        return response.ok({
-          body,
-        });
-      } catch (err) {
-        const error = transformError(err);
-        cspContext.logger.error(`Error while fetching status: ${err}`);
+const getHealthyAgents = async (
```

Comment:
I would prefer if we won't move utility functions in the file, it makes the diff much harder to read.
Also, its a subjective prefernce, and it would be inconvenient if every one of us will move functions to top or bottom when he changes stuff.

lastly, although possible, there's an eslint rule with good information and examples of why it's better to write the declarations before the usage and not the other way around.
https://eslint.org/docs/latest/rules/no-use-before-define

Reply:
So we've discussed this f2f and agreed not to change ordering of existing code
And I personally would change my preference to write in the order of exported last and keep the order of declaration